Authorization/Authorization/Presentation/SocialSign/SocialSignViewModel.swift

```diff
@@ -0,0 +1,128 @@
+//
+//  SocialSignViewModel.swift
+//  Authorization
+//
+//  Created by Eugene Yatsenko on 11.10.2023.
+//
+
+import SwiftUI
+import Core
+import AuthenticationServices
+import FacebookLogin
+import GoogleSignIn
+import MSAL
+import Swinject
+
+enum SocialResult {
+    case apple(AppleCredentials)
+    case facebook(LoginManagerLoginResult)
+    case google(GIDSignInResult)
+    case microsoft(MSALAccount, String)
+
+    var backend: String {
+        switch self {
+        case .apple:
+            return "apple-id"
+        case .facebook:
+            return "facebook"
+        case .google:
+            return "google-oauth2"
+        case .microsoft:
+            return "azuread-oauth2"
+        }
+    }
+}
+
+final public class SocialSignViewModel: ObservableObject {
+
+    // MARK: - Properties -
+
+    private var onSigned: ((Result<SocialResult, Error>) -> Void)
+
+    init(
+        config: ConfigProtocol = Container.shared.resolve(ConfigProtocol.self) ?? ConfigMock(),
```

Comment:
i think we should not provide default value from here in constructor, we should explicitly provide the config from outside.